tools/project-context.js

```diff
@@ -985,7 +1013,7 @@ _.extend(exports.ProjectConstraintsFile.prototype, {
   eachConstraint: function (iterator) {
     var self = this;
     _.each(self._constraintLines, function (lineRecord) {
-      if (lineRecord.constraint)
+      if (! lineRecord.skipOnRead && lineRecord.constraint)
```

Comment:
If `eachConstraint` ignores `extra-packages` maybe this needs to happen in `getConstraint` and the other functions too? So should this really be skipping in this case?